bin/github-deploy.sh

```diff
@@ -88,14 +88,34 @@ fi
 
 # Safety check, if a patch release is detected ask for verification.
 VERSION_PIECES=${VERSION//[^.]}
-if [[ "${#VERSION_PIECES}" -ge "2" && "$(echo "${DO_WP_DEPLOY:-n}" | tr "[:upper:]" "[:lower:]")" = "y" ]]; then
+
+# explode version parts
+split_version() {
+	echo ${VERSION} \
+	| sed 's/\./ /g'
+}
+
+SPLIT_VERSION=($(split_version))
+
+# IF VERSION_PIECES is less than 2 then its invalid so let's update it and notify
+if [[ "${#VERSION_PIECES}" -lt "2" ]]; then
+	if [[ ${#VERSION_PIECES} -eq "0" ]]; then
+		VERSION=${VERSION}.0.0
+	else
+		VERSION=${VERSION}.0
+	fi
+fi
+
+if [[ "${#VERSION_PIECES}" -ge "2" && "${SPLIT_VERSION[2]}" -ne "0" && "$(echo "${DO_WP_DEPLOY:-n}" | tr "[:upper:]" "[:lower:]")" = "y" ]]; then
 	output 1 "The version you entered (${VERSION}) looks like a patch version. Since this version will be deployed to WordPress.org, it will become the latest available version. Are you sure you want that (no will abort)?: [y/N]"
```

Comment:
The current configured workflow runs a WordPress deploy whenever a GitHub _release_ is created (differs from just a `tag`).  Earlier in the flow, the user is asked if this release is being deployed to WordPress. The problem that can happen is if the user indicates they are releasing a patch version to WordPress and we already have a higher version available on WordPress, this will result in the available version to be bumped down to the patch version.

So to be clear, this question only gets asked when:

- user has answered yes to deploying this on WordPress.org (Yes means a GitHub release gets created, no means just a GitHub tag is created).
- The version the user provided is detected as a patch version. It meets the criteria of `x.x.[>0]`

_So why ask the question, why not just abort?_

The question is asked because there _may_ be cases where the patch version _would_ be the latest version on WP.org and we _do_ want to release there.

What I can do is improve the first question about WordPress version and provide more context there (so the user knows when to answer yes or no).